### PR TITLE
block and report (wip)

### PR DESCRIPTION
**Pull request by KentoNishi**

### CODE REVIEW

**Comment by KentoNishi** [comment]
https://user-images.githubusercontent.com/38841491/167770103-c7d855be-5f5a-4c40-8d27-5d93693a6a83.mp4

**Comment by KentoNishi** [comment]
@r2dev2 @ChrRubin ***DO NOT USE THE REPORT FEATURE ON ARBITRARY COMMENTS!*** YT bans people who submit a bunch of false reports.

**Review by r2dev2 — CHANGES_REQUESTED** [review]
Blocking works in chrome, but not in firefox. In firefox, it errs out and the following is logged to the console:

firefox blocking

**Comment by KentoNishi** [comment]
i think i know what the issue is. fck

**Review by r2dev2 — APPROVED** [review]
Both blocking and reporting work on my end.

File: src/scripts/chat-interceptor.ts [hunk 1]
```diff
@@ -25,17 +26,25 @@ function injectedFunction(): void {
...
   };
+  window.dispatchEvent(new CustomEvent('chatLoaded', {
+    detail: JSON.stringify(window.ytcfg)
+  }));
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  window.addEventListener('proxyFetchRequest', async (event) => {
+    const args = JSON.parse((event as any).detail as string) as [string, object];
+    const request = await fetchFallback(...args);
+    const response = await request.json();
+    console.log(response);
```
Comment (ChrRubin) [inline src/scripts/chat-interceptor.ts:37]: We keeping this?

File: src/scripts/chat-background.ts [hunk 2]
```diff
@@ -275,16 +288,19 @@ const setInitialData = (port: Chat.Port, message: Chat.JsonMsg): void => {
...
 
-  const user =
-    (parsedJson?.continuationContents?.liveChatContinuation ||
-      parsedJson?.contents?.liveChatRenderer)
-      ?.actionPanel?.liveChatMessageInputRenderer
+  const actionPanel = (parsedJson?.continuationContents?.liveChatContinuation ||
+    parsedJson?.contents?.liveChatRenderer)
+    ?.actionPanel;
+
+  const user = actionPanel.liveChatRestrictedParticipationRenderer
```
Comment (ChrRubin) [inline src/scripts/chat-background.ts:295]: actionPanel can be undefined.

image

**Review by ChrRubin — CHANGES_REQUESTED** [review]
The menu should be hidden when user is not logged in. Both of the options will error when not logged in.

image

image

image

**Review by ChrRubin — APPROVED** [review]

**Comment by KentoNishi** [comment]
Let's fckn gooooooo

**Pull request merged**